show/main.py

```diff
@@ -768,23 +771,241 @@ def get_bgp_peer():
-def route(args, verbose):
+def route(args, namespace, display, verbose):
     """Show IP (IPv4) routing table"""
-    cmd = 'sudo vtysh -c "show ip route'
-
+    is_multi_asic = multi_asic.is_multi_asic()
+    device = multi_asic_util.MultiAsic(display, namespace)
+    arg_strg = ""
+    found_json = 0
+    hdr_printed = False
+    del_cnt = 0
+    if is_multi_asic:
+        # handling multi-ASIC by gathering the output from specified/all name space into a dictionary via
+        # jason option and then filter out the json entries (by removing those next Hop that are
+        # back-end ASIC if display is for front-end only). If the entry itself has no more next Hop after filtering,
+        # then skip over that particular route entry.  Once completed, if the user chose "json" option,
+        # then just print out the dictionary in Json format accordingly. But if no "json" option specified,
+        # then print out the header and the decoded entry representation for each route accordingly.
+        # This code is based on FRR 7.2 branch. If we moved to a new version we may need to change here as well
+        asic_cnt = multi_asic.get_num_asics()
+        if display == "all":
+            filter_back_end = False
+        else:
+            filter_back_end = True
+        if str(namespace) == "None":
+            start_range = 0
+            end_range = asic_cnt
+        else:
+            # -n asic0|asic1|...|asicN    So skip the first 4 chars to get the namespace number only
+            start_range = int(namespace[4:])
+            end_range = start_range + 1
+    else:
+        asic_cnt = 1
+        filter_back_end = False
+        start_range = 0
+        end_range = asic_cnt
+    # get all the other arguments except json that needs to be the last argument of the cmd if present
     for arg in args:
-        cmd += " " + str(arg)
-
-    cmd += '"'
-
-    run_command(cmd, display_cmd=verbose)
+        arg_strg += " " + str(arg)
+        if str(arg) == "json":
+            found_json = 1
+    if not found_json:
+        arg_strg += " json"
+
+    for ns in range(start_range, end_range):
+        filtered_route = {}
+        # Need to add "ns" to form bgpX so it is sent to the correct bgpX docker to handle the request
+        # If not MultiASIC, use space instad of ns number
+        if is_multi_asic:
+            ns_str = str(ns)
+        else:
+            ns_str = " "
+        cmd = "sudo docker exec bgp{0} vtysh -c 'show ip route {1}'".format(ns_str, arg_strg)
+        output = commands.getoutput(cmd)
+        # in case no output or something went wrong with user specified cmd argument(s) error it out
+        # error from FRR always start with character "%"
+        if output == "":
+            return
+        if output[0] == "%":
+            # remove the "json" keyword that was added by this handler so it wont confuse the user
+            error_msg = output[:-4]
```

Comment:
what if "json" keywork wasn't specified.

This can be removed if 'json' is passed as click option.

Reply:
See previous comment response in regard to this "optional" argument. which we intent to keep the same way as before.